File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/ScoreboardObjectivePacket.java
package gg.mineral.server.network.packet.play.clientbound;

import gg.mineral.api.network.packet.Packet;

import io.netty.buffer.ByteBuf;

public final record ScoreboardObjectivePacket(String objectiveName, String objectiveValue, byte createOrRemove)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        writeString(os, objectiveName);
        writeString(os, objectiveValue);
        os.writeByte(createOrRemove);
    }

    @Override
    public byte getId() {
        return 0x3B;
    }
}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/ScoreboardObjectivePacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class ScoreboardObjectivePacket(val objectiveName: String, val objectiveValue: String, val createOrRemove: Byte) :
    Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeString(objectiveName, objectiveValue)
        os.writeByte(createOrRemove.toInt())
    }

    override val id: Byte
        get() = 0x3B
}
